torch/onnx/symbolic_helper.py

```diff
@@ -148,36 +186,43 @@ def _maybe_get_scalar(value):
 
 def _get_const(value, desc, arg_name):
     if not _is_constant(value):
-        raise RuntimeError(
-            f"ONNX symbolic expected a constant value of the {arg_name} argument, "
-            f"got `{value}`"
+        raise errors.SymbolicValueError(
+            f"ONNX symbolic expected a constant value of the '{arg_name}' argument, "
+            f"got '{value}'",
+            value,
         )
     return _parse_arg(value, desc)
 
 
 def _unpack_list(list_value: _C.Value) -> List[_C.Value]:
     list_node = list_value.node()
-    assert list_node.kind() == "prim::ListConstruct"
+    if list_node.kind() != "prim::ListConstruct":
+        raise errors.SymbolicValueError(
+            f"ONNX symbolic expected node type prim::ListConstruct, "
+            f"got '{list_node}'.",
+            list_value,
+        )
     return list(list_node.inputs())
 
 
-def _unpack_tuple(tuple_value):
+def _unpack_tuple(tuple_value: _C.Value) -> Tuple[_C.Value, ...]:
     tuple_node = tuple_value.node()
     if tuple_node.kind() != "prim::TupleConstruct":
-        raise RuntimeError(
-            f"ONNX symbolic expected node type `prim::TupleConstruct`, "
-            f"got `{tuple_node}`"
+        raise errors.SymbolicValueError(
+            f"ONNX symbolic expected node type 'prim::TupleConstruct', "
+            f"got '{tuple_node.kind()}'.",
+            tuple_value,
         )
-    return list(tuple_node.inputs())
+    return tuple(tuple_node.inputs())
```

Comment:
I see this output being consumed in dequantized_helper with slicing. Is here changed to tuple a good idea?

Reply:
I was hoping to mark it as an immutable value. Confirmed offline that dequantized_helper is fine